pkg/dockerregistry/testutil/fakeopenshift.go

```diff
@@ -0,0 +1,271 @@
+package testutil
+
+import (
+	"fmt"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/client/testing/core"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	"github.com/docker/distribution/context"
+
+	"github.com/openshift/origin/pkg/client/testclient"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+// FakeOpenShift is an in-mempory reactors for fake.Client.
+type FakeOpenShift struct {
+	logger context.Logger
+
+	imagesStorage       map[string]*imageapi.Image
+	imageStreamsStorage map[string]*imageapi.ImageStream
+}
+
+// NewFakeOpenShift constructs the fake OpenShift reactors.
+func NewFakeOpenShift() *FakeOpenShift {
+	return &FakeOpenShift{
+		logger: context.GetLogger(context.Background()),
+
+		imagesStorage:       make(map[string]*imageapi.Image),
+		imageStreamsStorage: make(map[string]*imageapi.ImageStream),
+	}
+}
+
+// NewFakeOpenShiftWithClient constructs a fake client associated with
+// the stateful fake in-memory OpenShift reactors. The fake OpenShift is
+// available for direct interaction, so you can make buggy states.
+func NewFakeOpenShiftWithClient() (*FakeOpenShift, *testclient.Fake) {
+	os := NewFakeOpenShift()
+	client := &testclient.Fake{}
+	os.InjectTo(client)
+	return os, client
+}
+
+func (os *FakeOpenShift) CreateImage(image *imageapi.Image) (*imageapi.Image, error) {
+	_, ok := os.imagesStorage[image.Name]
+	if ok {
+		return nil, fmt.Errorf("image %s already exists", image.Name)
+	}
+
+	os.logger.Debug("(*FakeOpenShift).CreateImage: ", image.Name)
+	os.imagesStorage[image.Name] = image
+	return image, nil
+}
+
+func (os *FakeOpenShift) GetImage(name string) (*imageapi.Image, error) {
+	image, ok := os.imagesStorage[name]
+	if !ok {
+		return nil, fmt.Errorf("image %s not found", name)
```

Comment:
Ideally, the errors returned from inside of these reactors should be kerrors. [NewNotFound](https://github.com/openshift/origin/blob/cc52f421f8bf5cbd44bf5e57a59d8fde72edcdcd/vendor/k8s.io/kubernetes/pkg/api/errors/errors.go#L93) in this case.